middleware/categories.js

```diff
@@ -0,0 +1,10 @@
+import categories from '@/assets/js/categories'
+
+const categoryIds = categories.map(cat => cat.id)
+
+export default function(ctx) {
+  const category = ctx.params.category
+  if (category != null && !categoryIds.includes(ctx.params.category)) {
```

Comment:
`if (category != null && !categoryIds.includes(category)) {`

Reply:
What's more, shouldn't `if (!categoryIds.includes(category)) {` be fine?

Reply:
One question, what should happen if `category` is null?

Reply:
It will mean we are in the root. Category is the first uri segment.

Reply:
So, would it be okay to redirect this case to `/` too?

Reply:
It would be pointless to redirect from / to /, and maybe it could cause cycles.